app/filtering.js

```diff
@@ -372,7 +367,7 @@ function registerPermissionHandler (session, partition) {
       cb(result)
       if (persist) {
         // remember site setting for this host over http(s)
-        appActions.changeSiteSetting(urlOrigin, permission + 'Permission', result, isPrivate)
+        appActions.changeSiteSetting(getOrigin(url), permission + 'Permission', result, isPrivate)
```

Comment:
actually it looks like this change makes the permissions work by top-level origin instead of the origin asking for the permission; i'll fix to make it consistent with the way it was before.